core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/InfrastructureService.java

```diff
@@ -90,7 +99,42 @@ public List<Infrastructure> findAll(Predicate<Infrastructure> filter) {
 
   @PreAuthorize("hasPermission(#infrastructure, 'DELETE')")
   public void delete(Infrastructure infrastructure) {
-    throw new UnsupportedOperationException("Infrastructure deletion not currently supported.");
+    handlerService.handleDelete(infrastructure);
+    streamBindingView
+      .findAll(sb -> sb.getKey().getInfrastructureName().equals(infrastructure.getKey().getName()))
```

Comment:
Third time looking at this PR and just noticed, shouldn't this be comparing on the `InfrastructureKey`, not the Infrastructure name?

```suggestion
      .findAll(sb -> sb.getKey().getInfrastructureKey().equals(infrastructure.getKey()))
```

Reply:
The name and the zone together identify an Infra